SCClassLibrary/Common/Control/Server.sc

```diff
@@ -970,24 +975,141 @@ Server {
 					// not sure this should be here or elsewhere...
 					// this.bootInit(recover);
 					if(startAliveThread) { statusWatcher.start };
+					appBooted = true
 				};
 			} {
 				statusWatcher.serverBooting = false;
 				"%.boot - server process rebooting.\n".postf(this);
-				this.reboot;
+				fork { 0.01.wait; this.reboot };
 			}
 		}, {
 			this.bootServerApp({
 				if (Server.postingBootInfo) {
 					"%.boot - after bootServerApp...\n".postf(this);
 				};
-				if(startAliveThread) { statusWatcher.start }
-			}, )
+				if(startAliveThread) { statusWatcher.start };
+			})
 		}, 0.25);
+
+		// quick inlining of initial bootsteps
+		fork {
+			var test, tests = { [this.applicationRunning, this.hasBooted, this.notified] };
+			while {
+				test = tests.().every(_ == true);
+				test.not and: { timeout > 0 }
+			} {
+				timeout = timeout - 0.1;
+				0.1.wait;
+			};
+			if (test) {
+				this.prRunBootTask;
+			} {
+				"% boot failed after timeout!".format(this).warn;
+			}
+		}
+	}
+
+	bootAlt { | startAliveThread = true, recover = false, onFailure, onComplete, timeout = 5 |
```

Comment:
would this become the new `boot` method eventually?

Reply:
If agreed on, yes!